src/pages/index.js

```diff
@@ -1,140 +1,250 @@
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  cardWrap.prepend(card.getCardElement());
+  const card = createCard(cardData);
+  cardSection.addItem(card);
 }
-
+//Create Card Function
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteButton,
+    handleLikeButton
+  );
+  return card.getCardElement();
+}
+//Click Card Function
 function handleImageClick(link, name) {
   popupWithImage.open({ name, link });
 }
 
-// Form Validators
-const profileEditValidator = new FormValidator(config, profileEditForm);
-const addCardValidator = new FormValidator(config, addCardFormElement);
-profileEditValidator.enableValidation();
-addCardValidator.enableValidation();
+function handleEditProfileSubmit(inputValues) {
+  profileEditModal.renderLoading("Saving...");
+  api
+    .updateUserInfo(inputValues)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+      profileEditValidator.resetValidation();
+    })
+    .catch((err) => {
+      console.log(`Unable to process request, ${err}`);
+    })
+    .finally(() => {
+      profileEditModal.close();
+      profileEditModal.renderLoading("Save");
+    });
+}
+
+function handleAddCardSubmit(inputValues) {
+  addCardPopup.renderLoading("Loading...");
+  api
+    .createCard(inputValues)
+    .then((res) => {
+      renderCard(res);
+      addCardPopup.close();
+      addCardValidator.resetValidation();
+    })
+    .finally(() => {
+      addCardPopup.renderLoading("Save");
+    });
+}
+
+function handleDeleteButton(card) {
+  deleteImageModal.open();
+  deleteImageModal.handleDelete(() => {
+    deleteImageModal.renderLoading("Saving...");
+    api
+      .deleteCard(card.id)
+      .then(() => {
+        card.deleteCard();
+      })
+      .catch((err) => {
+        console.log(`Unable to process request, ${err}`);
+      })
+      .finally(() => {
+        deleteImageModal.close();
+        deleteImageModal.renderLoading("Yes");
+      });
+  });
+}
+
+function handleLikeButton(card) {
+  if (card.isLiked) {
+    api
+      .deleteLike(card.id)
+      .then(() => {
+        card.handleIsLiked(false);
+      })
+      .catch((err) => {
+        console.log(`Unable to process request, ${err}`);
+      });
+  }
+  if (!card.isLiked) {
+    api
+      .likeCard(card.id)
+      .then(() => {
+        card.handleIsLiked(true);
+      })
+      .catch((err) => {
+        console.log(`Unable to process request, ${err}`);
+      });
+  }
+}
 
-// Event Handlers
-function fillEditProfileForm() {
-  profileTitleInput.value = profileTitle.textContent;
-  profileDescriptionInput.value = profileDescription.textContent;
+function handleChangeProfilePicture(inputUrl) {
+  profileImageModal.renderLoading("Loading...");
+  api
+    .updateProfileImage(inputUrl)
+    .then((res) => {
+      userInfo.setProfileImage(res);
+      profileImageModal.close();
+      profileImageForm.resetValidation();
```

Comment:
Please, delete it

Method `resetValidation` is for clearing possible validation errors, when you open the **profile** form, because you insert **valid** data into its inputs. But when you submit a form, it's useless